include/seqan3/core/simd/simd_algorithm.hpp

```diff
@@ -36,6 +39,60 @@ constexpr simd_t iota_impl(scalar_t const offset, std::integer_sequence<scalar_t
     return simd_t{static_cast<scalar_t>(offset + I)...};
 }
 
+/*!\brief Transforms a batch of ranges into simd types (Structure-of-Arrays) and writes them into the output iterator.
+ * \tparam simd_t        The simd type; must model seqan3::simd::simd_concept.
+ * \tparam output_iter_t The output iterator type; must model std::OutputIterator with `simd_t`.
+ * \tparam range_t       The input range type; must model std::ranges::ForwardRange and its reference type must model
+ *                       std::ranges::ViewableRange and std::ranges::ForwardRange over seqan3::Alphabet.
+ * \param out     The output iterator.
+ * \param seq_rng The range over the ranges to transform.
+ * \ingroup simd
+ *
+ * \details
+ *
+ * This function assumes that the batch contains simd_traits<simd_t>::length many sequences which have the same
+ * size. At most simd_traits<simd_t>::length characters of the sequence are transformed or less if the ranges are
+ * smaller. If less sequences are provided or the sequences have different size the behaviour is undefined.
+ */
+template <simd_concept simd_t,
+          std::OutputIterator<simd_t> output_iter_t,
+          std::ranges::InputRange range_t>
+//!\cond
+    requires std::ranges::ForwardRange<reference_t<range_t>> &&
+             std::ranges::ViewableRange<reference_t<range_t>> &&
+             seqan3::Alphabet<reference_t<reference_t<range_t>>>
+//!\endcond
+constexpr void transform_batch_to_soa(output_iter_t out, range_t && seq_rng)
+{
+    // Check if the length is identical in debug mode.
+    if constexpr (std::ranges::SizedRange<range_t>)
+        assert(std::ranges::size(seq_rng) == static_cast<size_t>(simd_traits<simd_t>::length));
+
+    // stack memory for the cached iterator - sentinel pairs.
+    using it_pair_t = std::pair<std::ranges::iterator_t<reference_t<range_t>>,
+                                std::ranges::sentinel_t<reference_t<range_t>>>;
+    std::array<it_pair_t, simd_traits<simd_t>::length> iter_cache;
+
+    // Cache the iterator, sentinel pairs of the underlying ranges.
+    for (auto && [rng, idx] : std::view::zip(seq_rng, std::view::iota(0)))
+        iter_cache[idx] = {std::ranges::begin(rng), std::ranges::end(rng)};
```

Comment:
We should discuss this. This should be the input and you should not do this extra step in here.

Reply:
Ok, let's discuss this! Maybe today?